File: client/lib/domain/entities/post/post.dart
import 'package:client/domain/entities/api_object.dart';
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'post.freezed.dart';

part 'post.g.dart';

@freezed
class Post with _$Post implements ApiObject {
  const factory Post({
    required String id,
    required String userId,
    required String authorName,
    required String text,
    required String title,
    required DateTime date,
    @Default(0) int commentCount,
  }) = _Post;

  factory Post.fromJson(Map<String, Object?> json) => _$PostFromJson(json);
}

@freezed
class CreatePostModel with _$CreatePostModel implements ApiObject {
  const factory CreatePostModel({
    required String userId,
    required String authorName,
    required String text,
    required String title,
    required DateTime date,
    @Default(0) int commentCount,
  }) = _CreatePostModel;

  factory CreatePostModel.fromJson(Map<String, Object?> json) => _$CreatePostModelFromJson(json);
}

File: server/packages/rest/src/models/post.js
import mongoose from "mongoose";
const Schema = mongoose.Schema;

const post = new Schema({
  userId: {
    type: String,
    required: "user id is required",
  },
  authorName: {
    type: String,
    required: "author name is required",
  },
  title: {
    type: String,
    required: "title is required",
  },
  text: {
    type: String,
    required: "text is required",
  },
  date: {
    type: Date,
    default: Date.now,
  },
});

export default mongoose.model("posts", post);
